src/braket/circuits/circuit.py

```diff
@@ -458,6 +463,12 @@ def add_instruction(
         if target_mapping and target is not None:
             raise TypeError("Only one of 'target_mapping' or 'target' can be supplied.")
 
+        # Check if there is a measure instruction on the circuit
+        if not isinstance(instruction.operator, Measure) and any(
+            isinstance(instruction.operator, Measure) for instruction in self.instructions
+        ):
+            raise ValueError("Cannot add a gate or noise after a measure instruction.")
```

Comment:
Technically, we only need to prevent this in the case that `_measure_targets` intersects with `target` or `target_mapping.values()`. So this is more restrictive than it needs to be. However, if this an intentional choice for some reason (for simplicity of the code, etc.), I think that's fine - just wanted to call it out.

Reply:
That's a good call out and it is possible to make this less restrictive. One point for keeping the current error is that `add_instruction` can have no `target` and no `target_mapping` which would be a little more complex code-wise.

But, If you think it is more important to keep a customer use case, I can definitely change this.

Reply:
I don't think it's important - in most cases users would naturally place all the measurements at the end of the circuit. If we find that it is a problem, we can always tweak it later.

Reply:
> Technically, we only need to prevent this in the case that _measure_targets intersects with target or target_mapping.values().

Curious about how this works. For example:
```
with_measure = Circuit().h(0).cnot(0, 1).measure(1).h(0)
without_measure = Circuit().h(0).cnot(0, 1).h(0)
```
with_measure seems to put q0 in the |+> or |-> state with 50/50 odds, depending on the outcome of the measurement, while without the measure keeps q0 entangled with q1 in a more complex state. Are these two states indistinguishable, assuming we don't touch q1 anymore?

Reply:
Yes, they are indistinguishable in the sense that the measurement results will be unaffected. You're right that `measure(1)` causes the qubits to immediately become unentangled - but this only really matters if we're going to do subsequent operations which depend on the state of both qubits.

Another way of looking at it is that any single-qubit operation on qubit `i`, including measurement, commutes with any single-qubit operation on qubit `j`.

Reply:
(Note this is very similar to the reason that "faster-than-light communication" is impossible. There's no way for qubit 0 to "know" that a measurement was performed on qubit 1, unless some information is then explicitly transferred between them, e.g., a subsequent 2-qubit gate, or some kind of classical feed-forward that depends on the measurement result.)